MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -101,59 +43,80 @@ final class MovieQuizViewController: UIViewController {
-                                      message: model.text,
-                                      preferredStyle: .alert)
-        let action = UIAlertAction(title: model.buttonText, style: .default) { _ in
-            self.currentQuestionIndex = 0
-            self.correctAnswers = 0
-            let viewModel = self.convert(model: questions[self.currentQuestionIndex])
-            self.showNextQuestion(quiz: viewModel)
-        }
-        alert.addAction(action)
-        self.present(alert, animated: true)
-    }
-    
     private func showNextQuestionOrRoundResults() {
         imageView.layer.borderWidth = 0
-        if currentQuestionIndex == questions.count - 1 {
-            let viewModel = QuizResultsViewModel(
-                title: "Этот раунд окончен!",
-                text: "Ваш результат: \(correctAnswers)/10",
-                buttonText: "Сыграть ещё раз")
-            showRoundResult(model: viewModel)
+        if currentQuestionIndex == questionAmount - 1 {
+            guard let statisticService = statisticService else { return }
+            statisticService.store(correct: correctAnswers, total: questionAmount)
+            let date = statisticService.bestGame.date.dateTimeString
+            let message = """
+                Ваш результат: \(correctAnswers)/\(questionAmount)\n\
+                Количесство сыгранных квизов: \(statisticService.gamesCount)
+                Рекорд: \(statisticService.bestGame.correct)/\(statisticService.bestGame.total) (\(date))
+                Средняя точность: \(statisticService.totalAccuracy)
+            """
+            let alertModel = AlertModel(title: "Этот раунд окончен!",
+                                        message: message,
+                                        buttonText: "Сыграть ещё раз",
+                                        complection: { [weak self] in
+                                            guard let self = self else { return }
+                                            self.currentQuestionIndex = 0
+                                            self.correctAnswers = 0
+                                            self.questionFactory?.requestNextQuestion()
+                                        })
+            alertPresenter?.showAlert(model: alertModel)
         } else {
             currentQuestionIndex += 1
-            let viewModel = convert(model: questions[currentQuestionIndex])
-            showNextQuestion(quiz: viewModel)
+            questionFactory?.requestNextQuestion()
         }
     }
     
-    @IBAction private func noButtonClicked() {
-        let correctAnswer: Bool = questions[currentQuestionIndex].correctAnswer
-        showAnswerResult(isCorrect: correctAnswer == false)
+    @IBAction func noButtonClicked(_ sender: UIButton) {
+        guard let currentQuestion = currentQuestion else { return }
+        sender.isEnabled = false
+        let correctAnswer: Bool = currentQuestion.correctAnswer
```

Comment:
ты присваиваешь булево значение в константу - компилятор сам может вывести тип:
```suggestion
        let correctAnswer = currentQuestion.correctAnswer
```

Reply:
Это нужно для читаемости кода, чтобы другой разработчик не тратил время на то, чтобы понять типы данных

Reply:
Окей, принимается)

В некоторых командах можно встретить другое мнение: тип можно считать из грамотного нейминга (isCorrectAnswer точно не будет ничем иным, кроме как буля. Swift придумывали как лаконичный язык, и то что можно опустить - лучше опускать.

В любом случае, не настаиваю – твой стиль имеет право на жизнь и встречается в разработке на swift)